src/components/UploadingJobsDisplay/UploadingJobsDisplay.js

```diff
@@ -117,62 +171,129 @@ class UploadingJobsDisplay extends Component {
+  cancelCurrentFileUpload() {
+    if (this.currentFileUploadXhr) {
+      this.currentFileUploadXhr.abort();
+    }
+  }
+
+  async uploadFiles() {
+    const { files } = this.state;
+    const { setFiles } = this.context;
+
+    for (const fileKey of Object.keys(files)) {
+      try {
+        // cancel current and next file uploads if component is unmounted
+        if (!this._isMounted) {
+          this.cancelCurrentFileUpload();
+          setFiles(null);
+          break;
+        }
+
+        // eslint-disable-next-line no-await-in-loop
+        const response = await this.uploadFile(fileKey);
+
+        this.onFileUploadSuccess(response, fileKey);
+      } catch (error) {
+        this.onFileUploadFail(fileKey);
+      }
+    }
+
+    this.currentFileUploadXhr = null;
+    setFiles(null);
+  }
+
+  uploadFile(fileKey) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        this.currentFileUploadXhr = new XMLHttpRequest();
+
+        const { [fileKey]: fileMeta } = this.state.files;
+
+        const urlWithQueryParams = createUrl(this.fileUploaderUrl, {
+          fileId: fileMeta.id,
+          uploadDefinitionId: fileMeta.uploadDefinitionId,
+        });
 
+        this.currentFileUploadXhr.open('POST', urlWithQueryParams);
+        xhrAddHeaders(this.currentFileUploadXhr, this.createUploadFilesHeaders());
+
+        this.currentFileUploadXhr.upload.onprogress = event => {
+          this.onFileUploadProgress(fileKey, event);
+        };
+
+        this.currentFileUploadXhr.onreadystatechange = () => {
+          const {
+            status,
+            readyState,
+            response,
+          } = this.currentFileUploadXhr;
+
+          if (readyState !== 4) {
+            return;
+          }
+
+          if (status === 200) {
+            resolve(JSON.parse(response));
+          } else {
+            reject(JSON.parse(response));
+          }
+        };
+
+        this.currentFileUploadXhr.send(fileMeta.file);
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
+
+  createFilesDefinitionHeaders() {
     return {
+      ...createOkapiHeaders(this.props.stripes.okapi),
       'Content-Type': 'application/json',
-      'X-Okapi-Tenant': tenant,
-      'X-Okapi-Token': token,
     };
   }
 
   createUploadFilesHeaders() {
-    const {
-      token,
-      tenant,
-    } = this.props.stripes.okapi;
-
     return {
+      ...createOkapiHeaders(this.props.stripes.okapi),
       'Content-Type': 'application/octet-stream',
-      'Transfer-Encoding': 'chunked',
-      'X-Okapi-Tenant': tenant,
-      'X-Okapi-Token': token,
     };
   }
 
   createDeleteFileHeaders() {
-    const {
-      token,
-      tenant,
-    } = this.props.stripes.okapi;
-
-    return {
-      'X-Okapi-Tenant': tenant,
-      'X-Okapi-Token': token,
-    };
+    return createOkapiHeaders(this.props.stripes.okapi);
   }
 
-  prepareFiles(filesData) {
-    return filesData.reduce((res, file) => {
+  prepareFiles(files) {
```

Comment:
I propose to have the default parameter.

`(files = [])`

Reply:
Will not work if `null` is passed